### PR TITLE
Fix ENG-13271-check-as-begin-syntax

### PR DESCRIPTION
**Pull request by yzhang1991**
When splitting statements, we checked for keywords like "as", "begin", "case", "end" because they could change our semantics for a semicolon that we usually use to determine the end of a statement.

We used `Character.isLetterOrDigit` and `isLetterOrDigitFast` to check whether the token we checked was indeed a standalone token or a part of another token. But those two functions will return wrong answers when the tokens have "_" before or after them.

This pull request updates the functions we use so we can give right answers when underscores are presented.

### CODE REVIEW

File: src/frontend/org/voltdb/planner/ParsedSelectStmt.java [hunk 1]
```diff
@@ -829,16 +829,16 @@ private void parseDisplayColumn(int index, VoltXMLElement child,
...
-        }
         assert(colExpr != null);
 
         if (isDistributed) {
             colExpr = colExpr.replaceAVG();
             updateAvgExpressions();
         }
         ExpressionUtil.finalizeValueTypes(colExpr);
+        if (colExpr instanceof ConstantValueExpression) {
+            assert(colExpr.getValueType() != VoltType.NUMERIC);
```
Comment (wolffcm) [inline src/frontend/org/voltdb/planner/ParsedSelectStmt.java]: What's with this change?  Seems unrelated to everything else.

File: tests/frontend/org/voltdb/TestAdhocCreateStatementProc.java [hunk 2]
```diff
@@ -36,6 +36,28 @@
...
+    public void testENG13271() throws Exception {
+        String pathToCatalog = Configuration.getPathToCatalogForTest("adhocddl.jar");
+        VoltProjectBuilder builder = new VoltProjectBuilder();
+        builder.addLiteralSchema(
+            "CREATE TABLE T (\n" +
+            "   COLUMN_CASE tinyint\n" +
+            ");\n" +
+            "CREATE PROCEDURE PROC1\n" +
+            "AS BEGIN\n" +
+            "   INSERT INTO T (COLUMN_CASE) VALUES (?);\n" +
```
Comment (wolffcm) [inline tests/frontend/org/voltdb/TestAdhocCreateStatementProc.java]: So the issue was that they had `_case` in an identifier?  Should you be testing for `_` afterwards as well?

File: src/frontend/org/voltdb/planner/ParsedSelectStmt.java (hunk 1 above)
Comment (yzhang1991) [reply src/frontend/org/voltdb/planner/ParsedSelectStmt.java]: The user's query hit this with an assertion error because the `colExpr.getValueType() == VoltType.NUMERIC`. But this value type will be addressed by `ExpressionUtil.finalizeValueTypes`. So I am moving this assertion back a little bit.
It is really an issue if the expression is NUMERIC type even after being processed by the `finalizeValueTypes` function.

File: tests/frontend/org/voltdb/TestAdhocCreateStatementProc.java (hunk 2 above)
Comment (yzhang1991) [reply tests/frontend/org/voltdb/TestAdhocCreateStatementProc.java]: Adding a test case for it.

**Comment by yzhang1991** [comment]
The matchToToken function is only used for parsing multi-statement procedure DDL.
